File: Advanced C#/Homeworks/05. Regular Expressions Homework/06. Extract Hyperlinks/ExtractHyperlinks.cs
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections.Generic;
class ExtractHyperlinks
{
    static void Main()
    {
        // Input
        List<string> text = new List<string>();
        string currentLine = Console.ReadLine();
        while (currentLine.ToLower() != "end")
        {
            text.Add(currentLine + " ");
            currentLine = Console.ReadLine();
        }

        // Logic
        string pattern = @"(?<=href=([""'])).*?(?=\1)";
        Regex regex = new Regex(pattern);
        MatchCollection matches = regex.Matches(string.Join(" ", text));

        // Output
        foreach (Match match in matches)
        {
            Console.WriteLine(match);
        }
    }
}

File: JavaScript Basics/Homeworks/02. JavaScript Syntax/06. Exam Solutions/03. Extract Hyperlinks/extract-hyperlinks.js
function solve(args) {
    var input = args.join('\n');
    var regex = /<a\s+([^>]+\s+)?href\s*=\s*('([^']*)'|"([^"]*)|([^\s>]+))[^>]*>/g;

    var match;
    while(match = regex.exec(input)) {
        var hrefValue = match[3];
        if(hrefValue == undefined) {
            var hrefValue = match[4];
        }
        if(hrefValue == undefined) {
            var hrefValue = match[5];
        }
        console.log(hrefValue);
    }
}